File: codechef/july20/chfnswps.py
from collections import defaultdict

R = lambda: map(int, input().split())
t, = R()
for _ in range(t):
    n = R()
    map_a = R()
    map_b = R()
    a = list(map_a)
    b = list(map_b)
    min_a = min(a)
    min_b = min(b)
    all_min = min(min_a,min_b)
    all_count = defaultdict(int)
    A_count = defaultdict(int)
    B_count = defaultdict(int)
    for ele in a:
        all_count[ele] += 1
        A_count[ele]   += 1
    for ele in b:
        all_count[ele] += 1
        B_count[ele]   += 1

    flag = True
    A_ex = []
    B_ex = []
    for k,v in all_count.items():
        if(v%2==1):
            flag = False
            break
        temp_a = A_count[k] - v//2
        temp_b = B_count[k] - v//2

        if temp_a > 0:
            A_ex += temp_a * [k]
        elif temp_b > 0:
            B_ex += temp_b * [k]

    if not flag:
        print(-1)
    else:
        output = 0
        A_ex.sort()
        B_ex.sort()
        l = len(A_ex)
        # print(A_ex,B_ex)
        for i in range(l):
            output += min(2*all_min,A_ex[i],B_ex[l-i-1])
        print(output)

File: codechef/july20/chfnswps.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<map>
#include<cassert>
using namespace std;

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int t;
    cin >> t;
    while(t--){
        int n;
        cin >> n;
        map<long long int,int> A_count,B_count,total_count;
        long long int in;
        for(int i=0 ; i<n ; i++){
            cin >> in;
            total_count[in]++;
            A_count[in]++;
        }
        for(int i=0 ; i<n ; i++){
            cin >> in;
            total_count[in]++;
            B_count[in]++;
        }
        bool flag = true;
        vector<long long int> pos_a;
        vector<long long int> pos_b;
        for(const auto &kv: total_count){
            if(kv.second%2==1){
                flag = false;
                break;
            }
            else{
                int temp_a = A_count[kv.first] - kv.second/2;
                if(temp_a > 0){
                    while(temp_a--){
                        pos_a.push_back(kv.first);
                    }
                }
                int temp_b = B_count[kv.first] - kv.second/2;
                if(temp_b>0){
                    while(temp_b--){
                        pos_b.push_back(kv.first);
                    }
                }
            }
        }
        if(!flag) cout << -1 << "\n";
        else{
            long long int output = 0;
            sort(pos_a.begin(),pos_a.end());
            sort(pos_b.begin(),pos_b.end());
            int len = pos_a.size();
            assert(pos_a.size()==pos_b.size());
            for(int i=0 ; i<len ; i++){
                output += min(pos_a[i],pos_b[len-i-1]);
            }
            cout << output << "\n";
        }
    }
}
